src/screens/student_signup.tsx

```diff
@@ -1,11 +1,47 @@
 import { NativeStackScreenProps } from "@react-navigation/native-stack";
 import { Box, Button, Center, FormControl, Heading, Input, VStack } from "native-base";
-import React from "react";
+import React, { useEffect, useState } from "react";
+import { useQuery } from "react-query";
 import { LoginStackList } from "../components/login_stack";
+import { apiUrl } from "../constants";
 
 type StudentSignupProps = NativeStackScreenProps<LoginStackList, 'StudentSignup'>;
 
+type RegisterUserResponse = {
+  status: string
+};
+
 const StudentSignup = ({navigation}: StudentSignupProps) => {
+  const [firstName, setFirstName] = useState("");
+  const [lastName, setLastName] = useState("");
+  const [canonicalID, setCanonicalID] = useState("");
+  const [school, setSchool] = useState("");
+
+  const requestOptions = {
+    method: 'POST',
+    headers: { 
+      'Content-Type': 'application/json',
+      'Prefer': 'code=200',
+      'X-CSRF-TOKEN': '123',
+      'Authorization': 'Bearer SOMETOKENVALUE'
+    },
+    body: JSON.stringify({
+       type: 'Student',
+       firstName: firstName,
+       lastName: lastName,
+       canonicalID: canonicalID,
+       school: school,
+    })
+  };
+
+  const request = useQuery<RegisterUserResponse>('registerUser', async () => await (await fetch(apiUrl + '/student', requestOptions)).json(), {enabled: false});
+  useEffect(() => {
+    if (request.isSuccess) {
+      console.log(request.data.status);
+      if (request.data.status === "success") navigation.navigate('StudentSignup');
```

Comment:
Shouldn't this navigate to the home screen because the app is already of the student signup screen?